packages/react/src/merge-props/mergeProps.ts

```diff
@@ -29,11 +29,11 @@ type MergableProps<T extends React.ElementType> =
  * @param props props to merge.
  * @returns the merged props.
  */
-function mergeProps<T extends React.ElementType>(
-  ...props: MergableProps<T>[]
-): WithBaseUIEvent<React.ComponentPropsWithRef<T>> {
+export function mergeProps<Tag extends React.ElementType = 'div'>(
```

Comment:
I would make the generic required without default, it will force people to type it correctly.